pkg/helm/helmpath/default_home_dir.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright The Helm Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helmpath
+
+import (
+	"github.com/casimir/xdg-go"
+	"k8s.io/client-go/util/homedir"
+	"os"
+	"path/filepath"
+)
+
+// New default helm home, with different paths for different OS:
+//	- %APPDATA%\helm on Windows
+//	- ~/Library/Preferences/helm on OSX
+//  - $XDG_CONFIG_DIR/helm (typically ~/.config/helm for linux)
+var defaultHelmHome = xdg.App{Name: "helm"}.ConfigPath("")
```

Comment:
`pkg/helmpath` is purposely built around path traversal within `helm home`, but isn't intended to determine *where* `helm home` lives. Can you please move all of the logic surrounding `defaultHelmHome` out of `pkg/helmpath` and into `pkg/helm/environment`? Thanks.